mlos_bench/mlos_bench/service/config_persistence.py

```diff
@@ -108,6 +109,75 @@ def load_config(self, json_file_name: str) -> Union[dict, List[dict]]:
         with open(json_file_name, mode='r', encoding='utf-8') as fh_json:
             return json5.load(fh_json)  # type: ignore[no-any-return]
 
+    def prepare_class_load(self, config: dict,
+                           global_config: dict = None) -> Tuple[str, Dict[str, Any]]:
+        """
+        Extract the class instantiation parameters from the configuration.
+        Mix-in the global parameters and resolve the local file system paths,
+        where it is required.
+
+        Parameters
+        ----------
+        config : dict
+            Configuration of the optimizer.
+        global_config : dict
+            Global configuration parameters (optional).
+
+        Returns
+        -------
+        (class_name, class_config) : (str, dict)
+            Name of the class to instantiate and its configuration.
+        """
+        class_name = config["class"]
+        class_config = config.setdefault("config", {})
+
+        class_params = set(class_config)
+        for key in class_params.intersection(global_config or []):
+            class_config[key] = global_config[key]
+
+        for key in class_params.intersection(config.get("resolve_path", [])):
+            class_config[key] = self.resolve_path(class_config[key])
+
+        if _LOG.isEnabledFor(logging.DEBUG):
+            _LOG.debug("Instantiating: %s with config:\n%s",
+                       class_name, json.dumps(class_config, indent=2))
+
+        return (class_name, class_config)
+
+    def build_generic(self, cls, tunables: TunableGroups, service: Service,
+                      config: dict, global_config: dict = None):
+        # pylint: disable=too-many-arguments
```

Comment:
Pull the same `*` trick here too?

Reply:
oh that's right. great idea. I think that later I'll scan through all the code and see where else we have that pylint exception and fix it